### PR TITLE
Feature/aa 409 add additional data to analytics events

### PR DESCRIPTION
**Pull request by konstyefremov**

### CODE REVIEW

File: Api/Services/Analytics/AnalyticsService.cs [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-using HappyTravel.DataFormatters;
 using HappyTravel.MapperContracts.Internal.Mappings.Internals;
 using HappyTravel.MapperContracts.Public.Accommodations.Internals;
 using Accommodation = HappyTravel.MapperContracts.Public.Accommodations.Accommodation;
 
-namespace HappyTravel.Edo.Api.Services.Accommodations.Availability
+namespace HappyTravel.Edo.Api.Services.Analytics
 {
-    public class AvailabilityAnalyticsService
+    public class AnalyticsService
```
Comment (mingazhev) [inline Api/Services/Analytics/AnalyticsService.cs:18]: This is too common name for the service. We already have generic AnalyticsService unbound from the business logic.
If you'd like to use this service for availability + booking events, let's name it accordingly,
like `BookingAnalyticsService` which will include availability searches and bookings

File: Api/Services/Analytics/AnalyticsService.cs [hunk 2]
```diff
@@ -81,6 +81,30 @@ public void LogAccommodationAvailabilityRequested(in Accommodation accommodation
         }
         
         
+        public void LogBookingStatusChange(Booking booking,
```
Comment (mingazhev) [inline Api/Services/Analytics/AnalyticsService.cs:84]: This event looks too generic as I see.
It is not used for all status changes (which is correct), only for 2 of them. Why not using 2 separate events?
Saving this in a single event with unclear semantics will make reporting and analyzing harder as I see.

File: Api/Services/Analytics/AnalyticsService.cs (hunk 1 above)
Comment (konstyefremov) [reply Api/Services/Analytics/AnalyticsService.cs]: Fixed

File: Api/Services/Analytics/AnalyticsService.cs (hunk 2 above)
Comment (konstyefremov) [reply Api/Services/Analytics/AnalyticsService.cs]: Fixed.

File: Api/Models/Analytics/AgentAnalyticsInfo.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+namespace HappyTravel.Edo.Api.Models.Analytics
+{
+    public readonly struct AgentAnalyticsInfo
+    {
+        public AgentAnalyticsInfo(string counterpartyName)
+        {
+            CounterpartyName = counterpartyName;
+        }
+        
+        public string CounterpartyName { get; }
```
Comment (DFSko) [inline Api/Models/Analytics/AgentAnalyticsInfo.cs:10]: Do you really need a struct with only one property?

Comment (konstyefremov) [reply Api/Models/Analytics/AgentAnalyticsInfo.cs:10]: Yes. We discussed this with Vadim on Thursday. It most likely will be extended later with more properties.

File: Api/Services/Accommodations/Bookings/Management/BookingRecordsUpdater.cs [hunk 4]
```diff
@@ -158,22 +164,34 @@ async Task<Result> SendInvoice(AccommodationBookingInfo bookingInfo)
...
+
+
             void WriteFailureLog(string error) 
                 => _logger.LogBookingConfirmationFailure(booking.ReferenceCode, error);
         }
         
         
         private Task<Result> ProcessCancellation(Booking booking, DateTime cancellationDate, ApiCaller user)
         {
+            var counterParty = _context.Counterparties
```
Comment (mingazhev) [inline Api/Services/Accommodations/Bookings/Management/BookingRecordsUpdater.cs:178]: this is a single word, not two. `counterparty`

File: Api/Services/Analytics/PassengersInfo.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
+using System.Collections.Generic;
+using System.Linq;
+using HappyTravel.Edo.Data.Bookings;
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
+
+namespace HappyTravel.Edo.Api.Services.Analytics
+{
+    public readonly struct PassengersInfo
+    {
+        public PassengersInfo(List<BookedRoom> rooms)
```
Comment (mingazhev) [inline Api/Services/Analytics/PassengersInfo.cs:10]: You should avoid logic in constructors. Consider using static factory method to create the struct in this case

**Review by mingazhev — APPROVED** [review]
With small comments. May be merged after fixing

File: Api/Services/Analytics/PassengersInfo.cs (hunk 5 above)
Comment (konstyefremov) [reply Api/Services/Analytics/PassengersInfo.cs]: replaced with static factory method

File: Api/Services/Accommodations/Bookings/Management/BookingRecordsUpdater.cs (hunk 4 above)
Comment (konstyefremov) [reply Api/Services/Accommodations/Bookings/Management/BookingRecordsUpdater.cs]: fixed

**Pull request merged**